simphony/io/file_lattice.py

```diff
@@ -0,0 +1,194 @@
+            self._type = lattice.attrs.type
+            self._base_vect = lattice.attrs.base_vect
+            self._size = lattice.attrs.size
+            self._origin = lattice.attrs.origin
+
+    def get_node(self, index):
+        """Get a copy of the node corresponding to the given index.
+
+        Parameters:
+        -----------
+        index: tuple of D x int (node index coordinate)
+
+        Returns:
+        -----------
+        A reference to a LatticeNode object
+
+        """
+        n = self._get_row_number(index)
+        dcs = {}
+        colnum = 0
+        for column in xrange(self._table._table.description.mask.shape[0]):
+            if self._table._table[n][0][column]:
+                data = self._table._table.description.data
+                key = eval(CUBA.__name__+'.'+data._v_names[colnum].upper())
+                dcs[key] = self._table[n][key]
+            colnum += 1
+        return LatticeNode(index, dcs)
+
+    def update_node(self, node, keywords=None):
+        """ Updates FileLattice data for a LatticeNode
+
+        Parameters:
+            node : LatticeNode
+                reference to LatticeNode object
+            keywords : list, optional
+                List of CUBA-keywords that are updated. If empty list
+                then all DC's are from the node
+
+        """
+        # Find correct row for node
+        n = self._get_row_number(node.index)
+        colnms = self._table._table.description.data._v_names
+        if keywords is None:
+            # Update all table columns that are defined for node
+            ckeys = list({eval(CUBA.__name__+'.'+k.upper()) for k in colnms} &
+                         set(node.data.keys()))
+        else:
+            # Add only CUBA-keys that are listed in keywords and in Cudsfile
+            ckeys = list({eval(CUBA.__name__+'.'+k.upper()) for k in colnms} &
+                         set(keywords) & set(node.data.keys()))
+        self._table[n] = DataContainer({k: node.data[k] for k in ckeys})
+
+    def iter_nodes(self, indices=None):
+        """Get an iterator over the LatticeNodes described by the ids.
+
+        Parameters:
+        -----------
+        indices: iterable set of D x int, optional
+            node index coordinates
+
+        Returns:
+        -----------
+        A generator for LatticeNode objects
+
+        """
+        if indices is None:
+            for row in np.ndenumerate(self._table):
+                index = self._get_index(row[0])
+                yield self.get_node(index)
+        else:
+            for index in indices:
+                yield self.get_node(index)
+
+    def get_coordinate(self, index):
+        """Get coordinate of the given index coordinate.
+
+        Parameters:
+        -----------
+        index: D x int (node index coordinate)
+
+        Returns:
+        -----------
+        D x float
+
+        """
+        return self._origin + self._base_vect*np.array(index)
+
+    def _get_index(self, n):
```

Comment:
I think that these operations can be performed by numpy 
http://docs.scipy.org/doc/numpy/reference/generated/numpy.unravel_index.html
http://docs.scipy.org/doc/numpy/reference/generated/numpy.ravel_multi_index.html#numpy.ravel_multi_index